application/client/src/app/ui/views/toolbar/chart/chart/state.ts

```diff
@@ -0,0 +1,261 @@
+import { ISearchMap } from '@platform/interfaces/interface.rust.api.general';
+import { Instance as Logger } from '@platform/env/logger';
+import { Session } from '@service/session';
+import { Chart } from 'chart.js';
+import { StoredEntity } from '@service/session/dependencies/search/store';
+import { FilterRequest } from '@service/session/dependencies/search/filters/request';
+import { IPositionChange } from '@ui/service/toolbar/chart';
+import { Env, InternalAPI } from '@service/ilc';
+
+export interface IRange {
+    begin: number;
+    end: number;
+}
+
+export class State {
+    private _ilc!: InternalAPI;
+    private _env!: Env;
+    private _session!: Session;
+    private _logger!: Logger;
+    private _labels: string[] = [];
+    private _datasets: any[] = [];
+    private _map: ISearchMap = [];
+    private _detectChanges!: () => void;
```

Comment:
bellow you will find a comment, how to get access to all features of parent components (detectChages, ilc, etc)